jwst/ramp_fitting/ramp_fit_step.py

```diff
@@ -238,16 +440,63 @@ def process(self, input):
 
             int_times = input_model.int_times
 
+            # Before the ramp_fit() call, copy the input model ("_W" for weighting)
+            # for later reconstruction of the fitting array tuples.
+            input_model_W = copy.copy(input_model)
+
+            # Run ramp_fit(), ignoring all DO_NOT_USE groups, and return the
+            # ramp fitting arrays for the ImageModel, the CubeModel, and the
+            # RampFitOutputModel.
             image_info, integ_info, opt_info, gls_opt_model = ramp_fit.ramp_fit(
-                input_model, buffsize,
-                self.save_opt, readnoise_2d, gain_2d, self.algorithm,
-                self.weighting, max_cores, dqflags.pixel, suppress_one_group=self.suppress_one_group)
+                input_model, buffsize, self.save_opt, readnoise_2d, gain_2d,
+                self.algorithm, self.weighting, max_cores, dqflags.pixel,
+                suppress_one_group=self.suppress_one_group)
+
+            # Create a gdq to modify if there are undersampling-corrected groups
+            gdq = input_model_W.groupdq.copy()
+
+            # Locate groups where that are flagged with UNDERSAMP
+            wh_undersamp = np.where(np.bitwise_and(gdq.astype(np.int32), dqflags.group['UNDERSAMP']))
```

Comment:
Any particular reason why type `np.int32` is being used here, as opposed to the normal `np.uint32` for DQ flags?

Reply:
No, changed to np.uint32